### PR TITLE
[BugFix] QValue modules and nested action

### LINKED ISSUE #1273: [BUG] Is `_process_action_space_spec` compatible with nested keys?
Hello,

I am passing my nested action ("agents","action") spec to QValueModules and this is failing at

https://github.com/pytorch/rl/blob/main/torchrl/modules/tensordict_module/actors.py#L680

with  

"action could not be found in the spec. Make sure "
"you pass a spec that is either a native action spec or a composite action spec "
"with an 'action' entry. Otherwise, simply remove the spec and use the action_space only."

Is this intended?

### PR DESCRIPTION
**Pull request by matteobettini**
Fixes #1273

### CODE REVIEW

**Comment by matteobettini** [comment]
So @vmoens I have a proposition here.

This all business of allowing both `action_space` and `spec` to be specs is very confusing to me.

What about having `action_space` only be a key and `spec` only be a spec.

If so a user which wants to pass a str can pass it to `action_space`, but a user that passes a spec it passes it always to `spec`. 

Currently we allowed to feed a spec to either

**Comment by vmoens** [comment]
> So @vmoens I have a proposition here.
> 
> 
> 
> This all business of allowing both `action_space` and `spec` to be specs is very confusing to me.
> 
> 
> 
> What about having `action_space` only be a str and `spec` only be a spec.
> 
> 
> 
> If so a user which wants to pass a str can pass it to `action_space`, but a user that passes a spec passes it always to `spec`. 
> 
> 
> 
> Currently we allow to feed a spec to either

We can make the change but unfortunately as of now I'm not open to bc breaking changes without warning, ie if we do that we must ensure that users will only see a warning until 0.3 (since this will be in 0.2).

**Comment by matteobettini** [comment]
Sure, I can put the warning

File: torchrl/modules/tensordict_module/actors.py [hunk 1]
```diff
@@ -676,12 +676,22 @@ def _binary(self, value: torch.Tensor) -> torch.Tensor:
...
+        # this will break whenever our action is more complex than a single tensor
         try:
-            # this will break whenever our action is more complex than a single tensor
-            spec = spec["action"]
-            nest_action = True
+            spec_keys = spec.keys(True, True)
+            if "action" in spec_keys:
+                _key = "action"
+            else:
+                # the first key is the action
```
Comment (vmoens) [inline torchrl/modules/tensordict_module/actors.py:688]: getting the spec keys once and iterating twice can be dangerous is it's an iterable

[the remark this answers is not in the document]
Comment (vmoens) [reply torchrl/modules/tensordict_module/actors.py:688]: Do you think it makes sense? Do we want to be more restrictive?

**Review by vmoens — APPROVED** [review]
LGTM only some minor comments

File: torchrl/modules/tensordict_module/actors.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (matteobettini) [reply torchrl/modules/tensordict_module/actors.py:688]: We could only accept key that end with "action" as leaf.
up to you, i maybe slightly prefer the current version

**Pull request merged**